### PR TITLE
improve documentation about configuration

### LINKED ISSUE #349: Document CMD_S3_ENDPOINT ENV variable
The new environment variable``CMD_S3_ENDPOINT`` from #32 didn't get documented in the PR and will probably be missed by many people. Found by @oupala https://github.com/codimd/server/pull/32#issuecomment-621254222

### PR DESCRIPTION
**Pull request by oupala**
As discussed in #387 and with @davidmehren, here is a pull request to address some painpoints:

- I find it difficult to understand how to configure codimd, switching from environment variable to configuration file
  - there is also some inconsistency between both configuration solution (see #400 that fixes such inconsistency)
  - having to update two files while adding a configuration parameter is error prone (see #349 where a parameter hos not been documented)
- I am then proposing to merge two markdown files in one: configuration-config-file.md and configuration-env-vars.md
- I also fixed #349 and taken in account #400
- and finally I've added some linting default rule to have a consistent way of writing markdown

If the pull request seems ok, I'd like to imprive it by adding the default value of each parameter. This would ease the user experience while trying to figure out which parameter should be overidden or not. I've token about that in #348.

I'm now waiting for your feedback and I'm willing to discuss and change anything.

### CODE REVIEW

**Comment by ccoenen** [comment]
I think the `.remarkrc` should be discussed seperately. (I'm not opposed to it, I just think it should be a separate thing, maybe even including it in our lint / test run on CI if it's good :-D)

**Comment by ccoenen** [comment]
I'm somewhat reluctant about bunching the two files up into one, though. This was where we came from, and I don't think it helps readability to be honest. I would however wait for other opinions on this one.

**Comment by oupala** [comment]
As I have linted a markdown file, I wanted to join the linting rules that I used for linting, in order to avoid potential linting-wars.

But I completely agree that markdown linting is a much wider subject than this pull request. I'm ok to remove the commit about markdown linting and to open another pull request about that.

I'm just waiting for other opinions.

By the way, why is the merging blocked? Is there some manual reviewing of the travis build?

**Comment by SISheogorath** [comment]
For whatever reason the old travis integration is still coming up in some PRs. I haven't fully figured out yet, why this is the case. CI thinks your work is all good :)

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+### Lutim
+
+| config file | environment     | example value               | description                                                                 |
+| ----------- | --------------- | --------------------------- | --------------------------------------------------------------------------- |
+| `lutim`     |                 | `{"url": "YOUR_LUTIM_URL"}` | When `imageUploadType` is set to `lutim`, you can setup the lutim url       |
+|             | `CMD_LUTIM_URL` | `https://framapic.org/`     | When `CMD_IMAGE_UPLOAD_TYPE` is set to `lutim`, you can setup the lutim url |
+
+**Note:** *Due to the rename process we renamed all `HMD_`-prefix variables to be `CMD_`-prefixed. The old ones continue to work.*
+
+**Note:** *relative paths are based on CodiMD's base directory*
```
Comment (davidmehren) [inline docs/configuration.md:267]: Maybe move this to the 'CodiMD paths stuff' section as it seems to be most relevant there? And then remove the footnote superscripts in that section.

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+| ----------------------------- | ------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```
Comment (davidmehren) [inline docs/configuration.md:72]: This line seems to break the table rendering, at least in Githubs preview.
Is a horizontal line even necessary here?

Comment (oupala) [reply docs/configuration.md:72]: Note, this is a typo. I'll fix that.

File: docs/configuration.md (hunk 1 above)
Comment (oupala) [reply docs/configuration.md]: I moved the note about path from ` docs/configuration.md` to `docs/url-scheme.md` in b9c75a83f779a531bae666ce1b1afdb80cbee578.

I hope it was the file that you were talking about.

File: docs/configuration.md (hunk 2 above)
Comment (oupala) [reply docs/configuration.md]: Fixed in e46d520ca234c05cfe2431c5b7ab61b96194962e.

**Comment by oupala** [comment]
> I think the `.remarkrc` should be discussed seperately. (I'm not opposed to it, I just think it should be a separate thing, maybe even including it in our lint / test run on CI if it's good :-D)

I removed the `.remarkrc` file in order to discuss it in another pull request.

**Comment by oupala** [comment]
> I'm somewhat reluctant about bunching the two files up into one, though. This was where we came from, and I don't think it helps readability to be honest. I would however wait for other opinions on this one.

I'm not sure readability on the main point to look at. I think the current organization of the configuration document (2 separate files) is oriented on the **how** to configure codimd.

As a user, I was confronted to the problem of configuring codimd: some values can only be set by config file, some other can be configured by environment variables. It hard to guess what can be configured because of the way the documentation is organized: **how**. The content is splitted between two different files, for the same goal: configuring codimd.

With this pull request, I wanted to change the paradigm of the documentation to switch from **how** to configure codimd, to **what** can be configured on codimd.

Once you know **what** can be configured in codimd, each table tells you about **how** to configure this option.

I find this way of documenting more consistent for any user or admin of codimd.

In this comment, I wanted to explain why I created this pull request. Let discuss it further if needed.

File: docs/configuration.md (hunk 1 above)
Comment (davidmehren) [reply docs/configuration.md]: No, I was actually talking about [this section](https://github.com/oupala/server/blob/master/docs/configuration.md#L36) in the same file.

Comment (oupala) [reply docs/configuration.md]: Ok, fixed in 46e86b1be3e81956dd9e42604f370f380fa1b63b.

**Comment by SISheogorath** [comment]
Thank you very much!

**Pull request merged**